### PR TITLE
Adding default DQ flag mappings and IO methods

### PR DESCRIPTION
**Pull request by bmorris3**
### Description

This PR adds data quality flag definitions for JWST and Roman. This PR is independent from #2761, and can be merged before or after.

The flag definitions are written in `stdatamodels` ([see here](https://github.com/spacetelescope/stdatamodels/blob/41d4a43c1bec37d097a25866c1a76d93460ee306/src/stdatamodels/jwst/datamodels/dqflags.py#L26)) and `roman_datamodels` ([see here](https://github.com/spacetelescope/roman_datamodels/blob/62d7c96ea993bb354acf4b05e7e265ba0c7d32fa/src/roman_datamodels/dqflags.py#L27)) respectively. 

We could use the flag maps stored in these dependencies directly, since jdaviz already depends on `stdatamodels`, and `roman_datamodels` is an optional dependency. In this PR, I've chosen to save the definition tables as jdaviz package data in static CSV files. This ensures that they are impervious to upstream changes. While I don't expect that the _definitions_ are likely to change, the data structure that they're stored in, or their locations, may change. In this PR, I've added tests which compare the static CSV definitions with the upstream flag definitions in each package, to ensure going forward that both mappings contain all of the same flags, and the names for each flag match.

I've implemented `load_flag_map` to load flag maps from either a mission name ("jwst" or "roman") or a file path. I implemented `write_flag_map` so custom flag maps may be written to disk in the format that is expected by `load_flag_map`.

The tests do a spot-check on two flags in each mission, ensure that flag map load/write/load successfully round-trips, and ensure that the static CSV flag maps match their analogs in `stdatamodels` and `roman_datamodels`.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)? [🐱](https://jira.stsci.edu/browse/JDAT-4355)

### CODE REVIEW

File: jdaviz/configs/default/plugins/data_quality/dq_utils.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+        if flag_map_path is None:
+            raise NotImplementedError("No flag map found for "
+                                      f"mission/instrument '{mission_or_instrument}'.")
+
+    elif mission_or_instrument is None and path is not None:
+        # load directly from a CSV file:
+        flag_map_path = path
+
+    with resources.as_file(resources.files('jdaviz').joinpath(flag_map_path)) as path:
```
Comment (bmorris3) [inline jdaviz/configs/default/plugins/data_quality/dq_utils.py:39]: Note – we use `pkg_resources` to find line list data files within jdaviz. `pkg_resources` [has been deprecated](https://setuptools.pypa.io/en/latest/pkg_resources.html), so I'm giving the recommended alternatives a go (from `importlib`).

File: jdaviz/configs/default/plugins/data_quality/tests/test_data_quality.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+def test_flag_map_utils(mission_or_instrument, flag, name):
+    flag_map = load_flag_map(mission_or_instrument)
+    assert flag_map[flag]['name'] == name
+
+
+def test_load_write_load():
+    # load the JWST flag map
+    flag_map = load_flag_map('jwst')
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
```
Comment (pllim) [inline jdaviz/configs/default/plugins/data_quality/tests/test_data_quality.py:30]: Just happened to notice it scrolling by. Please use `tmp_path`. Thanks.

https://docs.pytest.org/en/stable/how-to/tmp_path.html

Comment (bmorris3) [reply jdaviz/configs/default/plugins/data_quality/tests/test_data_quality.py]: Done, thanks.

**Review by gibsongreen — APPROVED** [review]
If there's another commit from any of Clare's comments, I will take a second look. This is going to be an awesome plugin!

File: jdaviz/configs/default/plugins/data_quality/dq_utils.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    path : path-like
+        Path to a flag map file (CSV).
+
+    Returns
+    -------
+    flag_mapping : dict
+        Flag mapping definitions. Keys are powers of two,
+        values are the "name" and "description" for each flag.
+    """
+    if mission_or_instrument is not None and path is None:
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply jdaviz/configs/default/plugins/data_quality/dq_utils.py:28]: If this were commonly called yes, but I don't expect there will be many occasions where folks need to call these functions directly.

File: jdaviz/configs/default/plugins/data_quality/dq_utils.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    """
+    Write a table containing bitmask flags and their descriptions
+    to a CSV file.
+
+    Parameters
+    ----------
+    flag_mapping : dict
+        Flag mapping definitions. Keys are powers of two,
+        values are the "name" and "description" for each flag.
+    csv_path : path-like
```
[the remark this answers is not in the document]
Comment (bmorris3) [reply jdaviz/configs/default/plugins/data_quality/dq_utils.py:59]: As with the other comment, I don't think folks are likely to directly call these often, so I think this is ok.

**Pull request merged**